### DISCUSSION TITLE
What causes the 3D grid to display poorly in paraview?

### DISCUSSION THREAD

**Discussion by buzhiqiuyuebai** [SeisSol/SeisSol#824] [Q&A] [answered]
I directly open the loh3.surface.xdmf file to get a smooth waveform, but why is the waveform displayed on the surface obtained by extract surface so rough? Is it caused by the interpolation of refinement=3?
![image](https://user-images.githubusercontent.com/114474894/226297927-8ca8e9c7-93e3-4ffc-831d-1497186ef79e.png)
![image](https://user-images.githubusercontent.com/114474894/226298811-a8b64b69-30a5-4f01-aa2f-214bad1fc40b.png)

**Comment by Thomas-Ulrich** [comment, marked as answer]
Maybe what you see are Gibbs effects, more obvious because you sample the cells closer to their boundaries?

**Reply by buzhiqiuyuebai** [reply]
> Maybe what you see are Gibbs effects, more obvious because you sample the cells closer to their boundaries?

yes, this is a free surface, can I avoid the Gibbs effect by setting refinement=0? and why loh3.surface.xdmf doesn't have Gibbs effect? 
and in loh3.xdmf file, I can set stress and velocity by changing [parameters.par iOutputMask](https://github.com/SeisSol/Examples/blob/master/WP2_LOH1/parameters.par#:~:text=!%20%20%20%20%20%20%20%20%20%20%20%20%20%7Cstress%20%20%20%20%20%7Cvel,1%201%201)，in loh3.surface.xdmf how can I control the output and why there is three velocity in u and v (u1u2u3v1v2v3).
![image](https://user-images.githubusercontent.com/114474894/226310774-74298323-6892-4d66-8da9-65d70061c2ba.png)

**Reply by Thomas-Ulrich** [reply]
Gibbs effects are always there, but if you evaluate the quantities at the cell center (case of refinement=0), they are probably less pronounced than near the elements boundary. Probably it also depends on the order you use for your simulation.

**Reply by Thomas-Ulrich** [reply]
Honnestly, I have no idea why they would be only visible on the volume output, and not the free surface output.

**Reply by krenzland** [reply]
Could also be caused by the extract surface filter. At least in my experience, Paraview often makes some dubious choices when interpolating, leading to artifacts. Also, when using clip/similar things you need to be very careful, as you might cut through multiple elements. For clip, theres an option somewhere to NOT cut through tetrahedra. In this way, you'll get the correct data but not a smooth surface.

**Reply by buzhiqiuyuebai** [reply]
> Gibbs effects are always there, but if you evaluate the quantities at the cell center (case of refinement=0), they are probably less pronounced than near the elements boundary. Probably it also depends on the order you use for your simulation.

I did a [test](https://github.com/buzhiqiuyuebai/RefinementTest) with a simple example. When my grid is dense enough, even if I only use the second-order basis function to calculate, the waveform is very smooth. When I use the seventh-order basis function to calculate but the grid is sparse When using the strategy of refinement=3, there is a large Gibbs effect in the waveform, but for three-dimensional calculations, I think that the cost of improving the basis function is smaller than the cost of refining the grid while ensuring the calculation accuracy.

cl_1=25,SeisSol_Release_shsw_2_elastic，refinement=1
![image](https://user-images.githubusercontent.com/114474894/226539544-9257174c-cb3c-4332-ab7c-e99348e8969b.png)

cl_1=50,SeisSol_Release_shsw_2_elastic，refinement=1
![image](https://user-images.githubusercontent.com/114474894/226540428-9db0ce6c-40a2-4a5c-a3a9-3486d2db5308.png)

cl_1=75,SeisSol_Release_shsw_2_elastic，refinement=1
![image](https://user-images.githubusercontent.com/114474894/226539123-cd82dc8c-4545-4009-84a9-4ed4b3549e81.png)
cl_1=100, SeisSol_Release_shsw_2_elastic，refinement=1
![image](https://user-images.githubusercontent.com/114474894/226539798-5320f22b-0197-41e2-8c23-f368e384f89b.png)

cl_1=150, SeisSol_Release_shsw_7_elastic，refinement=3

![image](https://user-images.githubusercontent.com/114474894/226540299-88873466-7a63-4fde-a95b-7625a4f21bcf.png)

**Reply by Thomas-Ulrich** [reply]
I think Lukas is right, what we see are not Gibbs effect, but are the consequence of the fact that SeisSol paraview outputs have constant value within each cell (that is the value at the cell center).
For volume cells, cell centers are never aligned, even if they look aligned when you clip the mesh in paraview.

**Reply by krenzland** [reply]
There is a option which should be called "crincle slice" when using the slice/clip filter. If you try it, you know what I mean.

**Reply by buzhiqiuyuebai** [reply]
> I think Lukas is right, what we see are not Gibbs effect, but are the consequence of the fact that SeisSol paraview outputs have constant value within each cell (that is the value at the cell center). For volume cells, cell centers are never aligned, even if they look aligned when you clip the mesh in paraview.

I use a cube to test a point source with a wavelength size of 100 meters, when using a grid size of 75 meters and a basis function order of order 3; and when using a grid size of 150 meters and a basis function order of order 6 ; respectively get the following figure:

![image](https://user-images.githubusercontent.com/104721358/228609983-2c61dc19-4044-449a-a63c-09b94731a512.png)

![image](https://user-images.githubusercontent.com/104721358/228609782-679fc336-c588-4473-a2a7-48226bb1161e.png)

I placed the receiver at the same place and plotted its velocity component u with respect to the time transformation. It seems that the difference in the results does not seem to be very large, whether  is due to what you said `The fact that SeisSol paraview outputs have constant value within each cell (that is the value at the cell center).`  ???
As shown in the figure below, each tetrahedron shows a value, so that even if the calculation result is accurate enough, it does not look so accurate when displaying the volume output in paraview ???
![8538fabb8099b5ef37fe9daf19c3097](https://user-images.githubusercontent.com/114474894/228614729-231dd52a-55e0-469a-8954-8b2e31f79a52.png)
![image](https://user-images.githubusercontent.com/104721358/228611178-d8452373-86e8-4f7f-9de2-badba2d03f74.png)

**Reply by buzhiqiuyuebai** [reply]
> I think Lukas is right, what we see are not Gibbs effect, but are the consequence of the fact that SeisSol paraview outputs have constant value within each cell (that is the value at the cell center). For volume cells, cell centers are never aligned, even if they look aligned when you clip the mesh in paraview.
I am very sorry to bother you again on this issue, I would like to ask whether the calculated results stored in u.bin, v.bin, w.bin are already the central value of the tetrahedron, so even if I use the high-order method (For example, the seventh-order basis function), the calculated results can only be displayed in the center of the paraview, and can't be displayed with the values ​​of the vertices?

**Reply by Thomas-Ulrich** [reply]
there is no such option implemented yet, the paraview outputs are always evaluated at the cell or subcell centers.
